.buildkite/scripts/snyk/report.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+set -e
+
+TARGET_BRANCHES=("main")
+cd .buildkite/scripts
+
+# Resolves the branches we are going to track
+resolve_latest_branches() {
+  source snyk/resolve_stack_version.sh
+  for SNAPSHOT_VERSION in "${SNAPSHOT_VERSIONS[@]}"
+  do
+    IFS='.'
+    read -a versions <<< "$SNAPSHOT_VERSION"
+    version=${versions[0]}.${versions[1]}
+    version="${version%\"}"
+    version="${version#\"}"
+    TARGET_BRANCHES+=("$version")
+  done
+}
+
+# Clones the Logstash repo
+clone_logstash_repo() {
+  echo "Cloning logstash repo..."
+  git clone https://github.com/elastic/logstash.git
+}
+
+# Build Logstash specific branch to generate Gemlock file where Snyk scans
+build_logstash() {
+  cd logstash
+  git reset --hard HEAD # reset if any generated files appeared
+  git checkout "$1"
+  ./gradlew clean bootstrap assemble installDefaultGems && cd ..
+}
+
+# Downloads snyk distribution
+download_auth_snyk() {
+  cd logstash
+  echo "Downloading snyk..."
+  curl https://static.snyk.io/cli/latest/snyk-linux -o snyk
+  chmod +x ./snyk
+
+  vault_path=secret/ci/elastic-logstash-filter-elastic-integration/snyk-creds
+  SNYK_TOKEN=$(vault read -field=token "${vault_path}")
+  ./snyk auth "$SNYK_TOKEN"
+  cd ..
+}
+
+# Reports vulnerabilities to the Snyk
+report() {
+  echo "Reporting to Snyk..."
+  cd logstash
+  REMOTE_REPO_URL=$1
+  if [ "$REMOTE_REPO_URL" != "$MAIN_BRANCH" ]; then
+    MAJOR_VERSION=$(echo "$REMOTE_REPO_URL"| cut -d'.' -f 1)
+    REMOTE_REPO_URL="$MAJOR_VERSION".latest
+    echo "Using '$REMOTE_REPO_URL' remote repo url."
+  fi
+
+  # adding git add to Snyk tag to improve visibility
+  GIT_HEAD=$(git rev-parse --short HEAD 2> /dev/null | sed "s/\(.*\)/\1/")
+  ./snyk monitor --all-projects --org=logstash --remote-repo-url="$REMOTE_REPO_URL" --target-reference="$REMOTE_REPO_URL" --detection-depth=6 --exclude=requirements.txt --project-tags=branch="$TARGET_BRANCH",git_head="$GIT_HEAD" && true
```

Comment:
Review note: `exclude=requirements.txt` is belong to docker dependencies and we separately address container monitors.

Reply:
do we still want to continue in those situations?

Reply:
From the current situation it seems `yes` for me. Besides the Snyk service failures, as per my observations there are cases cause snyk process termination. Since we are using `--all-projects` option, snyk process will return exit code 2 when it cannot find `Gemlock` file under the project. For such cases we at least want to be sure we report successful monitors. Also, `snyk monitor` is in the loop, means if it gets terminated for `main`, we will be missing runs on `7.latest` & `8.latest`.